### DISCUSSION TITLE
Adding policy via JS

### DISCUSSION THREAD

**Discussion by MrVibe** [supabase/supabase#19984] [Questions] [answered]
I could not locate any documentation on how to add a policy on table and bucket programmatically. Any help / link towards documentation that I may have missed would be helpful.
In other case as this is a dire requirement of my project , can I use a service_role user to perform something like this :

`await supabase.from('pg_policies').upsert( ...    `

is this the right way ?

**Comment by GaryAustin1** [comment, marked as answer]
It would be a major security issue if postgres specific tables were accessible from the API and they are not.  The same applies to many Supabase specific tables like auth.

You would need to create policies with an rpc call to a security definer function and execute SQL to set up the policy.  You would still need to protect this function from being called by normal users by removing grants for anon, public and authenticated roles. 

As far as create a policy you can search on the web for ways to do that.  Certainly you could Execute SQL statement and alter the table to add them.  I personally am not familiar if you can just modify the pg_policies table.

https://supabase.com/docs/guides/database/functions#security-definer-vs-invoker
